src/http/request_handler.rs

```diff
@@ -173,6 +175,98 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &BTreeSet<String>) -> He
     new_headers
 }
 
+fn ensure_usable_cors_rules(layer: &CorsParams) {
+    if layer.allow_credentials {
+        assert!(
+            layer
+                .allow_headers
+                .as_ref()
+                .filter(|val| is_wildcard(val))
+                .is_none(),
+            "Invalid CORS configuration: Cannot combine `Access-Control-Allow-Credentials: true` \
+             with `Access-Control-Allow-Headers: *`"
+        );
+
+        assert!(
```

Comment:
Can't panic in request handlers. Move this to cors validation.

Reply:
fixed